packages/form-js-viewer/src/render/hooks/useValuesAsync.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'preact/hooks';
+import useService from './useService';
+
+/**
+ * @enum { String }
+ */
+export const LOAD_STATES = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error'
+};
+
+/**
+ * @typedef {Object} ValuesGetter
+ * @property {Object[]} values - The values data
+ * @property {(LOAD_STATES)} state - The values data's loading state, to use for conditional rendering
+ */
+
+/**
+ * A hook to load values for single and multiselect components.
+ *
+ * @param {Object} field - The form field to handle values for
+ * @return {ValuesGetter} valuesGetter - A values getter object providing loading state and values
+ */
+export default function(field) {
+  const {
+    valuesKey,
+    values: staticValues
+  } = field;
+
+  const [ valuesGetter, setValuesGetter ] = useState({ values: [], state: LOAD_STATES.LOADING });
+  const initialData = useService('form')._getState().initialData;
+
+  useEffect(() => {
+
+    let values = [];
+
+    if (valuesKey !== undefined) {
+
+      const safeInitialData = initialData || {};
+      const keyedValues = safeInitialData[ valuesKey ];
+
+      if (keyedValues && Array.isArray(keyedValues)) {
+        values = keyedValues;
+      }
+    }
+    else if (staticValues !== undefined) {
+      values = Array.isArray(staticValues) ? staticValues : [];
+    }
+    else {
+      setValuesGetter({ values, state: LOAD_STATES.ERROR });
```

Comment:
Is this expected in any case? If not, maybe we should even throw a proper error then.

If it's expected, maybe the `error` state alone is not enough context for the API consumer to handle this properly. At least an error message would be needed.

Reply:
And if it's expected, should we have a test case for this? E.g. testing what's happening when a form field neither defines static values nor dynamic input values.

Reply:
It's not expected in any case, except if someone messes with a form definition manually.

If they accidentally remove both, then push it to our ecosystem and it gets picked up by Tasklist. 

It might also hit if we load in that broken diagram into form-js. This scenario is worth testing. I know the properties panel is somewhat fault tolerant to it, but here we wouldn't want to be since we're in an unallowed design state. At the same time we also don't want to throw an actual exception since we'd still want people to be able to edit their way out of this problem via the properties panel. 
This is a linting concern also in a way. But we need some state to allow this to not crash the app but also make the user aware that something is wrong. I think we'll need to do more stuff with those error states in the future (for example if we're loading from an API, and it's down). So I would say, we keep this, but test it.